src/file.cc

```diff
@@ -266,6 +272,9 @@ void write_buffer_to_file(Buffer& buffer, StringView filename)
 
 void write_buffer_to_backup_file(Buffer& buffer)
 {
+    if (buffer.flags() & Buffer::Flags::ReadOnly)
+        return;
+
```

Comment:
Changes on readonly buffers are supposed to be temporary, so I disabled backups. It's arguable, maybe it's not the best way, but it was the simplest.